Tutorials/CNTK_209A_Berkeley_Segmentation_Dataset_loader.ipynb

```diff
@@ -0,0 +1,367 @@
+    "<p>We use similar reasoning for augmenting the training data here with <code>prep_224</code>. We will be selecting 224 x 224 patches and downscaling them to 112 x 112 patches. The only difference is that now we also rotate patches to additionally increase the number of training samples because we don't get as much 224 x 224 patches as 64 x 64 patches. Like before, 224 x 224 patches go into one folder and 112 x 112 patches go into another.</p>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {
+    "collapsed": true,
+    "deletable": true,
+    "editable": true
+   },
+   "outputs": [],
+   "source": [
+    "#extract 224 x 224 and 112 x 112 patches from BSDS dataset\n",
+    "def prep_224(images_dir, patch_h, patch_w, train112, train224):\n",
+    "    if not os.path.exists(train112):\n",
+    "        os.makedirs(train112)\n",
+    "    \n",
+    "    if not os.path.exists(train224):\n",
+    "        os.makedirs(train224)\n",
+    "\n",
+    "    k = 0\n",
+    "    \n",
+    "    print(\"Creating 224 x 224 and 112 x 112 training patches from:\", images_dir)\n",
+    "    \n",
+    "    for entry in os.listdir(images_dir):\n",
+    "        filename = os.path.join(images_dir, entry)\n",
+    "        img = Image.open(filename)\n",
+    "        rect = np.array(img)\n",
+    "    \n",
+    "        x = 0\n",
+    "        y = 0\n",
+    "    \n",
+    "        while(y + patch_h <= img.width):\n",
+    "            x = 0\n",
+    "            while(x + patch_w <= img.height):\n",
+    "                patch = rect[x : x + patch_h, y : y + patch_w]\n",
+    "                img_hr = Image.fromarray(patch, 'RGB')\n",
+    "                \n",
+    "                img_lr = img_hr.resize((patch_w // 2, patch_h // 2), Image.ANTIALIAS)\n",
+    "                \n",
+    "                for i in range(4):\n",
+    "                    out_hr = os.path.join(train224, str(k) + \".jpg\")\n",
+    "                    out_lr = os.path.join(train112, str(k) + \".jpg\")\n",
+    "                \n",
+    "                    k = k + 1\n",
+    "                \n",
+    "                    img_hr.save(out_hr)\n",
+    "                    img_lr.save(out_lr)\n",
+    "                \n",
+    "                    img_hr = img_hr.transpose(Image.ROTATE_90)\n",
+    "                    img_lr = img_lr.transpose(Image.ROTATE_90)\n",
+    "                \n",
+    "                x = x + 64\n",
+    "            y = y + 64\n",
+    "    print(\"Done!\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "deletable": true,
+    "editable": true
+   },
+   "source": [
+    "<p>We take the images we downloaded and run <code>prep</code> functions with adequate parameters. In the end, in the folder <b>Images</b> we will have a little bit less than 300 images from Berkeley Segmentation Dataset, depending on how many go to test set.</p>\n",
```

Comment:
We should also show a few image examples here (for example the original image and the resized image)